keras/layers/wrappers.py

```diff
@@ -184,6 +189,10 @@ def step(x, _):
            self.layer.activity_regularizer is not None):
             regularization_loss = self.layer.activity_regularizer(y)
             self.add_loss(regularization_loss, inputs)
+
+        if 'training' in kwargs:
+            y._uses_learning_phase = True
```

Comment:
This is incorrect in the general case: just because `call` takes a `training` argument doesn't mean the learning phase is being used. Instead, you should look at whether `_uses_learning_phase` is set on any if the output tensors of the child layer.